src/Hl7.Fhir.Base/Model/IScopedNode.cs

```diff
@@ -0,0 +1,255 @@
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Rest;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Transactions;
+
+namespace Hl7.Fhir.Model;
+
+#nullable enable
+
+[Flags]
+public enum NodeType
+{
+    Resource = 1,
+    Bundle = 1 << 1,
+    DomainResource = 1 << 2,
+    Primitive = 1 << 3,
+    Reference = 1 << 4,
+}
+
+/// <summary>
+/// An element within a tree of typed FHIR data with also a parent element.
+/// </summary>
+/// <remarks>
+/// This interface represents FHIR data as a tree of elements, including type information either present in
+/// the instance or derived from fully aware of the FHIR definitions and types
+/// </remarks>
+#pragma warning disable CS0618 // Type or member is obsolete
+public interface IScopedNode : ITypedElement, IShortPathGenerator
+#pragma warning restore CS0618 // Type or member is obsolete
+{
+    /// <summary>
+    /// The parent node of this node, or null if this is the root node.
+    /// </summary>
+    IScopedNode? Parent { get; }
+
+    /// <summary>
+    /// Enumerates the children of this instance. If name is given, only the children with that name are returned.
+    /// </summary>
+    /// <param name="name">If given, specifies the children to return based on this parameter.</param>
+    /// <returns>An IEnumerable containing the children of this instance, including scope data.</returns>
+    new IEnumerable<IScopedNode> Children(string? name = null);
+
+    /// <summary>
+    /// Name of the node, e.g. "active", "value".
+    /// </summary>
+    new string Name { get; }
```

Comment:
I used new for every property that will stop being defined in IScopedNode if we unlink it from ITypedElement. Since IScopedNode : ITypedElement and ITypedElement already has the property Name, we need new here.